### PR TITLE
Completed react-events exercise - Paxton Rosenberg

### PR DESCRIPTION
**Pull request by PaxtonRosenberg**
Screenshot 2023-12-22 at 3 06 22 PM

### CODE REVIEW

File: react-events/my-app/src/CustomButton.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+type Props = {
+  text: string;
+  color: string;
+  onCustomClick: (text: string) => void;
+};
+
+export default function CustomButton(props: Props) {
+  function handleClick() {
```
Comment (robgmerrill) [inline react-events/my-app/src/CustomButton.tsx:8]: This function is no longer needed. You are receiving the event handler as a prop so you can just use it within the JSX.

File: react-events/my-app/src/CustomButton.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    props.onCustomClick(props.text);
+  }
+  return (
+    <button style={{ backgroundColor: props.color }} onClick={handleClick}>
```
Comment (robgmerrill) [inline react-events/my-app/src/CustomButton.tsx:12]: You can now use the event handler you received as a prop directly.

`onClick={() => props.onCustomClick(props.text)}`

**Review by robgmerrill — CHANGES_REQUESTED** [review]
Close, but there is some extra code you no longer need from the first half of the exercise.

**Comment by PaxtonRosenberg** [comment]
Ah that makes a lot more sense - just corrected the excess code and removed some of the template styling. Also, de-structured props this time around. Let me know if this looks better.

**Review by robgmerrill — APPROVED** [review]
👍

**Pull request merged**